src/iceberg_manifest_list.cpp

```diff
@@ -0,0 +1,196 @@
+
+	//! Create the types for the DataChunk
+
+	child_list_t<Value> field_ids;
+	vector<string> names;
+	vector<LogicalType> types;
+
+	// manifest_path: string - 500
+	names.push_back("manifest_path");
+	types.push_back(LogicalType::VARCHAR);
+	field_ids.emplace_back("manifest_path", Value::INTEGER(MANIFEST_PATH));
+
+	// manifest_length: long - 501
+	names.push_back("manifest_length");
+	types.push_back(LogicalType::BIGINT);
+	field_ids.emplace_back("manifest_length", Value::INTEGER(MANIFEST_LENGTH));
+
+	// partition_spec_id: long - 502
+	names.push_back("partition_spec_id");
+	types.push_back(LogicalType::BIGINT);
+	field_ids.emplace_back("partition_spec_id", Value::INTEGER(PARTITION_SPEC_ID));
+
+	// content: int - 517
+	names.push_back("content");
+	types.push_back(LogicalType::INTEGER);
+	field_ids.emplace_back("content", Value::INTEGER(CONTENT));
+
+	// sequence_number: long - 515
+	names.push_back("sequence_number");
+	types.push_back(LogicalType::BIGINT);
+	field_ids.emplace_back("sequence_number", Value::INTEGER(SEQUENCE_NUMBER));
+
+	// min_sequence_number: long - 516
+	names.push_back("min_sequence_number");
+	types.push_back(LogicalType::BIGINT);
+	field_ids.emplace_back("min_sequence_number", Value::INTEGER(MIN_SEQUENCE_NUMBER));
+
+	// added_snapshot_id: long - 503
+	names.push_back("added_snapshot_id");
+	types.push_back(LogicalType::BIGINT);
+	field_ids.emplace_back("added_snapshot_id", Value::INTEGER(ADDED_SNAPSHOT_ID));
+
+	// added_files_count: int - 504
+	names.push_back("added_files_count");
+	types.push_back(LogicalType::INTEGER);
+	field_ids.emplace_back("added_files_count", Value::INTEGER(ADDED_FILES_COUNT));
+
+	// existing_files_count: int - 505
+	names.push_back("existing_files_count");
+	types.push_back(LogicalType::INTEGER);
+	field_ids.emplace_back("existing_files_count", Value::INTEGER(EXISTING_FILES_COUNT));
+
+	// deleted_files_count: int - 506
+	names.push_back("deleted_files_count");
+	types.push_back(LogicalType::INTEGER);
+	field_ids.emplace_back("deleted_files_count", Value::INTEGER(DELETED_FILES_COUNT));
+
+	// added_rows_count: long - 512
+	names.push_back("added_rows_count");
+	types.push_back(LogicalType::BIGINT);
+	field_ids.emplace_back("added_rows_count", Value::INTEGER(ADDED_ROWS_COUNT));
+
+	// existing_rows_count: long - 513
+	names.push_back("existing_rows_count");
+	types.push_back(LogicalType::BIGINT);
+	field_ids.emplace_back("existing_rows_count", Value::INTEGER(EXISTING_ROWS_COUNT));
+
+	// deleted_rows_count: long - 514
+	names.push_back("deleted_rows_count");
+	types.push_back(LogicalType::BIGINT);
+	field_ids.emplace_back("deleted_rows_count", Value::INTEGER(DELETED_ROWS_COUNT));
+
+	// partitions: list<508: field_summary> - 507
+	names.push_back("partitions");
+	types.push_back(FieldSummaryType());
+	field_ids.emplace_back("partitions", FieldSummaryFieldIds());
+
+	//! Populate the DataChunk with the manifests
+
+	DataChunk data;
```

Comment:
Future optimization: we can probably construct the data chunk once per transaction, even if we have multiple added snapshots.

The final manifest list is just a combination of diffs of the previous alters, by setting the cardinality lower we can use this chunk for previous alters (in the same transaction)